applications/tests/test_reservation_creation.py

```diff
@@ -0,0 +1,245 @@
+            "times": [
+                TimeElement(
+                    start_time=datetime.time(hour=0, tzinfo=get_default_timezone()),
+                    end_time=datetime.time(hour=18, tzinfo=get_default_timezone()),
+                    end_time_on_next_day=False,
+                )
+            ],
+        },
+    ]
+
+    return response
+
+
+@mock.patch(
+    "applications.utils.reservation_creation.get_opening_hours",
+    wraps=get_opening_hour_data_single_day,
+)
+class ReservationSchedulerTestCase(TestCase, DjangoTestCase):
+    @classmethod
+    def setUpTestData(cls) -> None:
+        date = datetime.date.today()
+        cls.round = ApplicationRoundFactory(
+            reservation_period_begin=date,
+            reservation_period_end=date + datetime.timedelta(weeks=4),
+        )
+        cls.application = ApplicationFactory(application_round=cls.round)
+        cls.application_event = ApplicationEventFactory(
+            application=cls.application,
+            events_per_week=1,
+            begin=date,
+            end=date + datetime.timedelta(weeks=4),
+        )
+        cls.res_unit = ReservationUnitFactory(spaces=[SpaceFactory()])
+        EventReservationUnitFactory(
+            application_event=cls.application_event,
+            reservation_unit=cls.res_unit,
+        )
+        cls.schedule = ApplicationEventScheduleFactory(
+            application_event=cls.application_event
+        )
+
+        cls.result = ApplicationEventScheduleResultFactory(
+            application_event_schedule=cls.schedule,
+            allocated_reservation_unit=cls.res_unit,
+            allocated_begin=datetime.time(hour=12),
+            allocated_end=datetime.time(hour=14),
+        )
+        cls.service_sector_manager = User.objects.create(
+            username="service_sector_person",
+            first_name="ser",
+            last_name="vice",
+            email="ser.vice.person@foo.com",
+        )
+        reservation_date = next_or_current_matching_weekday(
+            cls.application_event.begin, cls.result.allocated_day
+        )
+        cls.reservation_begin = datetime.datetime.combine(
+            reservation_date, cls.result.allocated_begin, tzinfo=get_default_timezone()
+        )
+        cls.reservation_end = datetime.datetime.combine(
+            reservation_date, cls.result.allocated_end, tzinfo=get_default_timezone()
+        )
+
+    def test_reservation_time_equal_reservation_time_when_open_in_reservation_time(
+        self, mock
+    ):
+        scheduler = ReservationScheduler(
+            self.res_unit.id, self.reservation_begin, self.reservation_end
+        )
+        start_dt, end_dt = scheduler.get_reservation_times_based_on_opening_hours()
+
+        assert_that(start_dt).is_equal_to(self.reservation_begin)
+        assert_that(end_dt).is_equal_to(self.reservation_end)
+
+    def test_reservation_time_is_one_hour_reduced_from_reservation_time_when_closes_before_reservation_end(
```

Comment:
Yes, that might be the case. I thought that the customer would have the possiblity to use the space at least some time if it wouldn't fit within the opening hours since allocation won't allocate to same time afaik?

Reply:
Ok, looks like this is good as it is.